index.js

```diff
@@ -12,47 +13,36 @@ function handleOpts(url, opts) {
   return opts;
 }
 
-var psi = module.exports = function (url, opts, cb) {
-  if (typeof opts !== 'object') {
-      cb = opts;
+var psi = module.exports = function (url, opts) {
+  return new Promise(function (resolve, reject) {
+    if (typeof opts !== 'object') {
       opts = {};
-  }
-
-  if (!url) {
-    throw new Error('URL required');
-  }
-
-  if (typeof cb !== 'function') {
-    throw new Error('Callback required');
-  }
+    }
 
-  pagespeed(handleOpts(url, opts), function (err, response) {
-    if (err) {
-      err.noStack = true;
-      cb(err);
-      return;
+    if (!url) {
+      throw new Error('URL required');
     }
 
-    cb(null, response);
+    pagespeed(handleOpts(url, opts), function (err, response) {
```

Comment:
You can use [pify](https://github.com/sindresorhus/pify) here and anywhere else where there are callback-style APIs.